jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ResourceSelector.kt

```diff
@@ -3,146 +3,122 @@
-            val previousState = this.isEnabled
-            loadingStatus = ResourceLoadingStatus.LOADING
-            isEnabled = previousState
-
-            val value = try {
-                fetch().apply {
-                    loadingException = null
-                    loadingStatus = ResourceLoadingStatus.SUCCESSFUL
-                }
-            } catch (e: Exception) {
-                LOG.warn(e) { "Failed to load values" }
-                loadingException = e
-                loadingStatus = ResourceLoadingStatus.FAILED
-                null
-            } ?: return@executeOnPooledThread
-
-            runInEdt(ModalityState.any()) {
-                val model = this.model as DefaultComboBoxModel<T>
-                model.addElement(value)
-                model.selectedItem = value
-                if (updateStatus) {
-                    this.isEnabled = true
-                } else {
-                    this.isEnabled = shouldBeEnabled
-                }
-            }
+    @Synchronized
+    override fun setEnabled(enabled: Boolean) {
+        shouldBeEnabled = enabled
+        when {
+            loadingStatus == Status.LOADED || !enabled -> super.setEnabled(enabled)
+            loadingStatus == Status.NOT_LOADED -> reload()
         }
     }
 
-    fun toValidationInfo(loading: String = message("loading_resource.loading"), failed: String = message("loading_resource.failed"), notSelected: String): ValidationInfo? {
-        if (this.selected() != null) {
-            return null
+    @TestOnly
+    fun forceLoaded() {
+        loadingStatus = Status.LOADED
+    }
+
+    private fun processSuccess(value: Collection<T>, previouslySelected: Any?) {
+        loadingStatus = Status.LOADED
+        runInEdt(ModalityState.any()) {
+            setEditable(false)
+            actualModel.removeAllElements()
+            value.sortedBy { it.toString().toLowerCase() }.forEach { actualModel.addElement(it) }
+            super.setEnabled(shouldBeEnabled)
+            super.setSelectedItem(determineSelection(selector, previouslySelected))
         }
-        // Error messages do not work on a disabled component. May be a JetBrains bug?
-        // Revisit after https://github.com/aws/aws-toolkit-jetbrains/issues/726
-        val component = if (super.isEnabled()) this else null
-        return when (loadingStatus) {
-            ResourceLoadingStatus.LOADING -> ValidationInfo(loading, component)
-            ResourceLoadingStatus.FAILED -> ValidationInfo(loadingException?.message ?: failed, component)
-            ResourceLoadingStatus.SUCCESSFUL -> ValidationInfo(notSelected, component)
+    }
+
+    private fun processFailure(error: Throwable) {
+        loadingStatus = Status.NOT_LOADED
+        val message = message("loading_resource.failed")
+        LOG.warn(error) { message }
+        runInEdt(ModalityState.any()) {
+            super.setSelectedItem(message)
+            toolTipText = error.message
```

Comment:
This affects the tooltips we use to indicate what a field does, are there any alternatives?

Reply:
fixed